server/src/routes/api/v1/sitesRouter.js

```diff
@@ -1,26 +1,60 @@
 import express from "express";
+import uploadImage from "../../../services/uploadImage.js";
+import cleanUserInput from "../../../services/cleanUserInput.js";
 import { Site } from "../../../models/index.js";
+import { ValidationError } from "objection";
+import SiteSerializer from "../../../serializers/SiteSerializer.js"
 
 const sitesRouter = new express.Router();
 
 sitesRouter.get("/", async (req, res) => {
     try {
         const sites = await Site.query();
-        return res.status(200).json({ siteList: sites });
+        
+        const serializedSites = sites.map(site => {
+            return SiteSerializer.getSummary(site)
+        })
+        return res.status(200).json({ siteList: serializedSites });
     } catch (error) {
         return res.status(500).json({ errors: error });
     }
 });
 
+sitesRouter.post("/", uploadImage.single("image"), async (req, res) => {
+    try {
+        const { body } = req;
+        const formInput = cleanUserInput(body);
+        let data
+        if (req.file) {
+            data = {
+                ...formInput,
+                image: req.file.location,
+            };
+        } else {
+            data = {
+                ...formInput,
+                image: "https://static.displate.com/857x1200/displate/2020-03-12/c47b057f270b9101cfb4d462279d38b3_7477322f46cfe492f40beb04fe6d42ff.jpg",
+            };
+        }
+        const newSite = await Site.query().insertAndFetch(data);
+        return res.status(201).json({ site: newSite });
```

Comment:
your post request on the front end is issuing a redirect away from the form, but generally we want to serialize any data being returned as JSON (don't need to add here, but something to keep in mind for other POST endpoints)